Generator/Sources/NeedleFramework/Entry/Generator.swift

```diff
@@ -120,11 +122,11 @@ public class Generator {
     
     private let sourceKitUtilities: SourceKitUtilities
 
-    private func createExecutor(withName name: String, shouldTrackTaskId: Bool) -> SequenceExecutor {
+    private func createExecutor(withName name: String, shouldTrackTaskId: Bool, concurrencyLimit: Int?) -> SequenceExecutor {
         #if DEBUG
             return ProcessInfo().environment["SINGLE_THREADED"] != nil ? SerialSequenceExecutor() : ConcurrentSequenceExecutor(name: name, qos: .userInteractive, shouldTrackTaskId: shouldTrackTaskId)
```

Comment:
This DEBUG path seems not so useful any more

Reply:
Serial is probably a bad name now. A better one would be something like Immediate. There is still the difference that the Concurrent executor would bounce threads, whereas the "Serial" one just executes immediately on the caller thread.

Reply:
@rudro wrote the original serial executor. I'll let him chime in on whether we should keep/rename it, or just delete it?

Reply:
Yes, I agree we may need to rename `Serial` to `SerialAndOnCurrentThread`.  Also, because of the small difference Yi mentioned it might still be better to have the same thread version just for ease of debugging.

Reply:
Ok. I'll rename the "serial" one to "immediate".

Reply:
Here's the rename. https://github.com/uber/swift-concurrency/pull/23